nvme: MAINT: Refactor telemetry-log command for build warning #1323

ikegami-t opened the pull request:
Signed-off-by: Tokunori Ikegami <ikegami.t@gmail.com>

ikegami-t commented:
Sorry fixed the commit message subject error and also separated additionally the function.

igaw commented on common.h:
```diff
@@ -33,4 +33,6 @@ static inline uint64_t mmio_read64(void *addr)
 	return ((uint64_t) high << 32) | low;
 }
 
+#define fallthrough do {} while (0)
```
All relative modern [compilers](https://gcc.gnu.org/onlinedocs/gcc/Statement-Attributes.html) know about fallthrough attribute/statement. Hence you should test if the compiler supports it.

The Linux kernel used to do something like this
```
#if __has_attribute(__fallthrough__)
# define fallthrough                    __attribute__((__fallthrough__))
#else
# define fallthrough                    do {} while (0)  /* fallthrough */
#endif
```
but IIRC it was now dropped as all supported gcc/clang versions know about fallthrough

igaw commented on nvme.c:
```diff
@@ -407,20 +407,110 @@ static int get_ana_log(int argc, char **argv, struct command *cmd,
-static int get_telemetry_log(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+static int get_telemetry_log_total_size_dalb4(int fd,
+					      struct nvme_telemetry_log *log,
+					      unsigned *total_size)
```
Could you also change from `unsigned` to `unsigned int` when touching this code?

ikegami-t replied:
Fixed as mentioned.

igaw commented on nvme.c:
```diff
@@ -469,84 +559,19 @@ static int get_telemetry_log(int argc, char **argv, struct command *cmd, struct
+	err = get_telemetry_log_total_size(fd, cfg.data_area, log, &total_size);
+	if (total_size) {
+		err = write(output, (void *)log, total_size);
+		if (err != total_size)
```
Since err and total_size have two different types, it's probably better to do:
`if (err < 0 || err != total_size)`

ikegami-t replied:
Fixed this also.

ikegami-t commented:
Thasnk for your comments. Fixed the code as you mentioned so please review again.

ikegami-t commented:
Additionally changed the code as below.
1. Changed "unsigned" still remained to "unsigned  int".
2. Changed "long long unsigned int" to "unsigned long long".

igaw commented:
Looks good. I just splited the type changes out of the initial patch.

igaw commented:
Thanks!
